biothings/hub/dataload/dumper.py

```diff
@@ -296,18 +296,28 @@ def register_status(self, status, transient=False, **extra):
                 )
                 self.src_doc.pop(field)
 
-        self.src_doc.update(
-            {
-                '_id': self.src_name,
-                'download': {
-                    'release': release,
-                    'data_folder': data_folder,
-                    'logfile': self.logfile,
-                    'started_at': datetime.now().astimezone(),
-                    'status': status,
-                },
-            }
-        )
+        current_download_info = {
+            "_id": self.src_name,
+            "download": {
+                "release": release,
+                "data_folder": data_folder,
+                "logfile": self.logfile,
+                "started_at": datetime.now().astimezone(),
+                "status": status,
+            },
+        }
+        # Update last success download time.
+        # If current status is success, we will get the current's started_at
+        # If failed, we will get the last_success from the last download instead
+        last_download_info = self.src_doc.get("download", {})
+        current_download_info["download"]["last_success"] = last_download_info.get("last_success")
```

Comment:
@sengineer0 will this line raises an error if `last_download_info` is empty (e.g. first download or last download failed)?

Reply:
No, it will not. `last_download_info` is set default to empty dict, and  `last_success` field will return None if it does not exist in `last_download_info`.

Reply:
In case if there is no information we want (e.g. first download or last download failed), this code will not failed, and the `current_download_info["download"]["last_success"]` will be set to None.